streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -1180,6 +1182,68 @@ void shutdown(final boolean clean) {
         }
     }
 
+    private void shutdownStateUpdater() {
+        if (stateUpdater != null) {
+            stateUpdater.shutdown(Duration.ofMillis(Long.MAX_VALUE));
```

Comment:
This is a meta thought: I'm not sure if passing in Long.MAX_VALUE is the best thing to do. When we shutdown a stream thread we do have a timeout at the shutdown thread waiting for those threads to join, and I'm thinking maybe we should propagate that timeout from StreamThread.completeShutdown -> TaskManager -> StateUpdater (and similarly for all other modules of a stream thread that we are shutting down).